packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2604,50 +2607,45 @@ class RenderTransform extends RenderProxyBox {
     );
   }
 
+  @override
+  OffsetLayer updateCompositedLayer({required covariant ImageFilterLayer? oldLayer}) {
+    final ImageFilterLayer layer = oldLayer ?? ImageFilterLayer();
+    layer.imageFilter = ui.ImageFilter.matrix(
+      _effectiveTransform!.storage,
+      filterQuality: filterQuality!
+    );
+    return layer;
+  }
+
   @override
   void paint(PaintingContext context, Offset offset) {
-    if (child != null) {
-      final Matrix4 transform = _effectiveTransform!;
-      if (filterQuality == null) {
-        final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
-        if (childOffset == null) {
-          // if the matrix is singular the children would be compressed to a line or
-          // single point, instead short-circuit and paint nothing.
-          final double det = transform.determinant();
-          if (det == 0 || !det.isFinite) {
-            layer = null;
-            return;
-          }
-          layer = context.pushTransform(
-            needsCompositing,
-            offset,
-            transform,
-            super.paint,
-            oldLayer: layer is TransformLayer ? layer as TransformLayer? : null,
-          );
-        } else {
-          super.paint(context, offset + childOffset);
-          layer = null;
-        }
-      } else {
-        final Matrix4 effectiveTransform = Matrix4.translationValues(offset.dx, offset.dy, 0.0)
-          ..multiply(transform)..translate(-offset.dx, -offset.dy);
-        final ui.ImageFilter filter = ui.ImageFilter.matrix(
-          effectiveTransform.storage,
-          filterQuality: filterQuality!,
-        );
-        if (layer is ImageFilterLayer) {
-          final ImageFilterLayer filterLayer = layer! as ImageFilterLayer;
-          filterLayer.imageFilter = filter;
-        } else {
-          layer = ImageFilterLayer(imageFilter: filter);
-        }
-        context.pushLayer(layer!, super.paint, offset);
-        assert(() {
-          layer!.debugCreator = debugCreator;
-          return true;
-        }());
+    if (child == null) {
+      return;
+    }
+    if (isRepaintBoundary) {
+      return super.paint(context, offset);
+    }
+
+    final Matrix4 transform = _effectiveTransform!;
+    final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
+    if (childOffset == null) {
```

Comment:
It feels like the new direction of this method is to trap cases that have a simple solution and do an early return as per the previous 2 tests (and the one that follows in the body of the if clause here). Would this fit in better with:

```
  if (childOffset != null) {
    super.paint(child, offset + childOffset);
    layer = null;
    return;
  }
```

instead of the if/else?

Reply:
Good idea, done